token/program-2022/src/instruction.rs

```diff
@@ -1744,6 +1773,21 @@ pub fn initialize_non_transferable_mint(
     })
 }
 
+/// Creates an `InitializePermanentDelegate` instruction
+pub fn initialize_permanent_delegate(
+    token_program_id: &Pubkey,
+    mint_pubkey: &Pubkey,
+    delegate: Option<&Pubkey>,
+) -> Result<Instruction, ProgramError> {
+    check_program_account(token_program_id)?;
+    let delegate = delegate.cloned().into();
+    Ok(Instruction {
+        program_id: *token_program_id,
+        accounts: vec![AccountMeta::new(*mint_pubkey, false)],
+        data: TokenInstruction::InitializePermanentDelegate { delegate }.pack(),
+    })
+}
+
```

Comment:
is there any reason to allow the user to init this as `None`? since theres no functional difference between not having the extension at all. (if we end up rejecting this, should also change `process_initialize_permanent_delegate` to reject it)

Reply:
That's a good point. I was thinking for maximum flexibility to do this, but it's kind of a silly case to support. I'll make it mandatory everywhere